packages/web/src/common/store/wallet/selectors.ts

```diff
@@ -1,19 +1,30 @@
-import { BNWei } from 'common/models/Wallet'
+import { createSelector } from '@reduxjs/toolkit'
+
+import { StringWei } from 'common/models/Wallet'
 import { CommonState } from 'common/store'
 import { Nullable } from 'common/utils/typeUtils'
 import { stringWeiToBN } from 'common/utils/wallet'
 
-export const getAccountBalance = (state: CommonState): Nullable<BNWei> => {
-  const balance = state.wallet.balance
-  if (!balance) return null
-  return stringWeiToBN(balance)
+// Previously, the getAccountBalance selector would return different
+// values (although shallowly the same) because of the return of a
```

Comment:
interesting find 👀. Wonder how many other places are affected by this...

nit: probably want to say something like "numerically the same" - shallow compare means checking reference equality, so in this case they wouldn't be shallowly the same